bmcd/src/flash_service.rs

```diff
@@ -0,0 +1,191 @@
+use crate::into_legacy_response::LegacyResponse;
+use actix_web::{http::StatusCode, web::Bytes};
+use anyhow::Context;
+use futures::future::BoxFuture;
+use std::{
+    collections::hash_map::DefaultHasher,
+    error::Error,
+    fmt::Display,
+    hash::{Hash, Hasher},
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::sync::mpsc::{channel, error::SendError, Sender};
+use tokio_util::sync::CancellationToken;
+use tpi_rs::app::flash_application::flash_node;
+use tpi_rs::{app::bmc_application::BmcApplication, middleware::NodeId, utils::logging_sink};
+use tpi_rs::{app::flash_application::FlashContext, utils::ReceiverReader};
+
+pub type FlashDoneFut = BoxFuture<'static, anyhow::Result<()>>;
+const RESET_TIMEOUT: Duration = Duration::from_secs(10);
+
+struct TransferContext {
+    pub peer: u64,
+    pub bytes_sender: Sender<Bytes>,
+    pub cancel: CancellationToken,
+    last_recieved_chunk: Instant,
+}
+
+impl TransferContext {
+    pub fn duration_since_last_chunk(&self) -> Duration {
+        Instant::now().saturating_duration_since(self.last_recieved_chunk)
+    }
+}
+
+pub struct FlashService {
+    status: Option<TransferContext>,
+    bmc: Arc<BmcApplication>,
+}
+
+impl FlashService {
+    pub fn new(bmc: Arc<BmcApplication>) -> Self {
+        Self { status: None, bmc }
+    }
+
+    pub async fn start_transfer(
+        &mut self,
+        peer: &str,
+        filename: String,
+        size: usize,
```

Comment:
Same as in tpi changes, this shouldn't be usize to handle files > 4 GiBs

Reply:
will fix